wbnf/parser.go

```diff
@@ -486,33 +486,39 @@ func (t Delim) Parser(rule Rule, c cache) parser.Parser {
 }
 
 type lrtgen struct {
-	sides   [2]Term
+	parent  *Term
+	sides   [2]*Term
```

Comment:
nit: The following feels cleaner. It avoids the cost of indexed lookups and doesn't require the `side` variable:
```go
	curSIde, nextSide
...
	out = l.curSide
	l.curSide, l.nextSide = l.nextSide, l.curSide
```